### PR TITLE
avoid adding extra text / trailing whitespaces in urls

### PR DESCRIPTION
**Pull request by jspaaks**
**Related issues**

Refs:

- N/A

**Describe the changes made in this pull request**

This PR updates the regex pattern for `url`s to be more strict about trailing characters (notably whitespace).

**Review checklist**

- [x] Please check if the pull request is against the correct branch  
(format/schema/semantic documentation changes: `develop`; typos, meta files, etc.: `main`)
- [x] Please check if all changes are recorded in `CHANGELOG.md` and adapt if necessary.
- [x] Please run tests locally.

```bash
cd $(mktemp -d --tmpdir cff.XXXXXX)
git clone https://github.com/citation-file-format/citation-file-format .
git checkout <branch>
python3 -m venv env
source env/bin/activate
pip install --upgrade pip wheel setuptools
pip install -r requirements.txt
pytest
<do other things>
```

- [ ] `<do other things>`

### CODE REVIEW

**Review by sdruskat — COMMENTED** [review]
Tested with 3 different online JSON Schema converters, and this works fine. Thanks, @jspaaks!

- Should the schema guide mention that trailing whitespaces are now disallowed?
- I'll update the changelog accordingly in this branch.

**Comment by jspaaks** [comment]
I'll update the schema guide with a note about the expected format

**Comment by jspaaks** [comment]
https://github.com/citation-file-format/citation-file-format/pull/468/commits/b58833b0cfc8e7aaaa721c7aa21e5a96882cbd42 added a note about whitespace

**Comment by jspaaks** [comment]
This PR could still use a test under 1.3.0 once PR #470 is merged

**Comment by jspaaks** [comment]
> Should the schema guide mention that trailing whitespaces are now disallowed?

FYI: The note I added (https://github.com/citation-file-format/citation-file-format/commit/b58833b0cfc8e7aaaa721c7aa21e5a96882cbd42) has been superseded by simply referring to `strictish-string`

File: CHANGELOG.md [hunk 1]
```diff
@@ -18,6 +18,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - Added more license identifiers from SPDX. Issue [#454](https://github.com/citation-file-format/citation-file-format/issues/454); PR [#461](https://github.com/citation-file-format/citation-file-format/pull/461)
 - Added machine readable relation qualifiers to `identifiers` using terminology from DataCite v4.4; PR [#459](https://github.com/citation-file-format/citation-file-format/pull/459)
 - Fixed: instances of `person` now explicitly require one of `alias`, `email`, `given-names`, `family-names` or `orcid` to avoid accidentally having empty person objects (notably, `authors`). PR [#462](https://github.com/citation-file-format/citation-file-format/pull/462)
+- Changed the regular expression for URLs to disallow trailing whitespaces. PR [#468](https://github.com/citation-file-format/citation-file-format/pull/468)
```
Comment (sdruskat) [inline CHANGELOG.md:21]: This is duplicated in #504. I like the other entry better, so this could go?

```suggestion
```

**Review by sdruskat — APPROVED** [review]
Thanks!

One comment re the changelog entry which is duplicated in the summary changelog PR #504, feel free to heed or dismiss, good to go either way :).

File: CHANGELOG.md (hunk 1 above)
Comment (jspaaks) [reply CHANGELOG.md:21]: Agreed

**Pull request merged**